src/pages/institution/institution-solution/institution-solution.ts

```diff
@@ -0,0 +1,266 @@
+      this.submissionArr.includes(sol.submissionId)
+        ? (sol.downloaded = true)
+        : null;
+    });
+  }
+
+  goToEcm(solutionIndex, submissionId) {
+    // let submissionId = id;
+    let heading = this.institutionList.entities[this.entityType][
+      this.entityIndex
+    ].name;
+    let recentlyUpdatedEntity = {
+      programName: this.institutionList.entities[this.entityType][
+        this.entityIndex
+      ].solutions[solutionIndex].programName,
+      ProgramId: this.institutionList.entities[this.entityType][
+        this.entityIndex
+      ].solutions[solutionIndex].programId,
+      EntityName: this.institutionList.entities[this.entityType][
+        this.entityIndex
+      ].name,
+      EntityId: this.institutionList.entities[this.entityType][this.entityIndex]
+        ._id,
+      submissionId: submissionId,
+    };
+    console.log("go to ecm called" + submissionId);
+
+    this.localStorage
+      .getLocalStorage(this.utils.getAssessmentLocalStorageKey(submissionId))
+      .then((successData) => {
+        console.log(JSON.stringify(successData));
+        //console.log("go to ecm called");
+
+        // successData = this.updateTracker.getLastModified(successData , submissionId)
+        console.log("after modification");
+        if (successData.assessment.evidences.length > 1) {
+          this.navCtrl.push("EvidenceListPage", {
+            _id: submissionId,
+            name: heading,
+            recentlyUpdatedEntity: recentlyUpdatedEntity,
+          });
+        } else {
+          if (successData.assessment.evidences[0].startTime) {
+            //console.log("if loop " + successData.assessment.evidences[0].externalId)
+            this.utils.setCurrentimageFolderName(
+              successData.assessment.evidences[0].externalId,
+              submissionId
+            );
+            this.navCtrl.push("SectionListPage", {
+              _id: submissionId,
+              name: heading,
+              selectedEvidence: 0,
+              recentlyUpdatedEntity: recentlyUpdatedEntity,
+            });
+          } else {
+            const assessment = {
+              _id: submissionId,
+              name: heading,
+              recentlyUpdatedEntity: recentlyUpdatedEntity,
+            };
+            this.openAction(assessment, successData, 0);
+            //console.log("else loop");
+          }
+        }
+      })
+      .catch((error) => {});
+  }
+
+  openAction(assessment, aseessmemtData, evidenceIndex) {
+    this.utils.setCurrentimageFolderName(
+      aseessmemtData.assessment.evidences[evidenceIndex].externalId,
+      assessment._id
+    );
+    const options = {
+      _id: assessment._id,
+      name: assessment.name,
+      recentlyUpdatedEntity: assessment.recentlyUpdatedEntity,
+      selectedEvidence: evidenceIndex,
+      entityDetails: aseessmemtData,
+    };
+    this.evdnsServ.openActionSheet(options);
+  }
+
+  getAssessmentDetails(solutionIndex) {
```

Comment:
Can we reuse this in programs and the institution's flow?

Reply:
same reason as the structure is different i am using two different functions to manipulate the data in each flows